code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -222,35 +285,38 @@
 		return
 
 	if(accept_check(O))
-		if(contents.len >= max_n_of_items)
-			user << "<span class='notice'>\The [src] is full.</span>"
-			return 1
-		else
-			user.remove_from_mob(O)
-			O.loc = src
-			if(item_quants[O.name])
-				item_quants[O.name]++
-			else
-				item_quants[O.name] = 1
-			user.visible_message("<span class='notice'>[user] has added \the [O] to \the [src].</span>", "<span class='notice'>You add \the [O] to \the [src].</span>")
+		user.remove_from_mob(O)
+		var/hasRecord = FALSE	//Check to see if this passes or not.
+		for(var/datum/data/stored_item/I in item_records)
+			if(istype(O, I.item_path))
+				stock(O, I)
+				qdel(O)
+				return
+		if(!hasRecord)
+			var/datum/data/stored_item/item = new/datum/data/stored_item(src, O.type)
+			stock(O, item)
+			item_records.Add(item)
+			
+		user.visible_message("<span class='notice'>[user] has added \the [O] to \the [src].</span>", "<span class='notice'>You add \the [O] to \the [src].</span>")
 
-			nanomanager.update_uis(src)
+		nanomanager.update_uis(src)
 
 	else if(istype(O, /obj/item/weapon/storage/bag))
 		var/obj/item/weapon/storage/bag/P = O
 		var/plants_loaded = 0
 		for(var/obj/G in P.contents)
 			if(accept_check(G))
-				if(contents.len >= max_n_of_items)
-					user << "<span class='notice'>\The [src] is full.</span>"
-					return 1
-				else
-					P.remove_from_storage(G,src)
-					if(item_quants[G.name])
-						item_quants[G.name]++
-					else
-						item_quants[G.name] = 1
-					plants_loaded++
+				plants_loaded++
+				var/hasRecord = FALSE	//Check to see if this passes or not.
+				for(var/datum/data/stored_item/I in item_records)
+					if(istype(G, I.item_path))
+						stock(G, I)
+						hasRecord = TRUE
+				if(!hasRecord)
+					var/datum/data/stored_item/item = new/datum/data/stored_item(src, G.type)
+					stock(G, item)
+					item_records.Add(item)

```

Comment:
Very similar to the insertion done above, easily made into a proc to share the code.